### ISSUE TITLE
soil-clean-up causing Uncaught SyntaxError: Unexpected number

### ISSUE THREAD

**Issue by ekandreas** [roots/soil#238]
# Description

When using theme support soil-clean-up the browser responds Uncaught SyntaxError: Unexpected number

**Comment by ecksite** [comment]
I also get an error that seems to be related to the soil-clean-up.

In my case the console shows the following error, when in Wordpress Customizer:
I can fix this by uncommenting this part in `soil/modules/clean-up.php`

```php
/**
 * Clean up output of <script> tags
 */
function clean_script_tag($input) {
  $input = str_replace("type='text/javascript' ", '', $input);
  return str_replace("'", '"', $input);
}
add_filter('script_loader_tag', __NAMESPACE__ . '\\clean_script_tag');
 ```

To be exact, it seems that the `str_replace("'", '"', $input);` line, causes the issue.

Its hard to debug, and I can not tell wether this is related to my environment setup or to the replacement at the moment. I believe this happened the first time, after I updated to Wordpress 5.3 (roots/wordpress).

**Comment by DaveLak** [comment]
I ran into this same issue when Core tries to include polyfills on the front end. Specifically, the following script tag was the causing the error:

```html
<script type="text/javascript">
( "fetch" in window ) || document.write( "<script src="/wp-includes/js/dist/vendor/wp-polyfill-fetch.min.js"></scr" + "ipt>" );( document.contains ) || document.write( "<script src="/wp-includes/js/dist/vendor/wp-polyfill-node-contains.min.js"></scr" + "ipt>" );( window.FormData && window.FormData.prototype.keys ) || document.write( "<script src="/wp-includes/js/dist/vendor/wp-polyfill-formdata.min.js"></scr" + "ipt>" );( Element.prototype.matches && Element.prototype.closest ) || document.write( "<script src="/wp-includes/js/dist/vendor/wp-polyfill-element-closest.min.js"></scr" + "ipt>" );
</script>
```

The error was the result of the `str_replace` inside `Roots\Soil\CleanUp\clean_script_tag()` as @ecksite mentioned. In my case the single quotes (`'`) surrounding script tags inside the  the inside the `document.write()` calls was being replaced with double quotes (`"`), meaning the value of the `src` attributes was no longer treated as part of the string.

As a temporary fix I've removed `clean_script_tag` with the following:

```php
add_action('init', function () {
    remove_filter('script_loader_tag', 'Roots\\Soil\\CleanUp\\clean_script_tag');
});
```

This fixes the JS syntax error in my case.

**Comment by retlehs** [comment]
closed by #224
